### ISSUE TITLE
429 error coming from ES rest api not handled

### ISSUE THREAD

**Issue by terebentina** [elastic/elasticsearch-js#232]
I am getting some 429 (too many requests) coming from elasticsearch api. However, elasticsearch-js reports this as 'unknown error'. I had to console log the error body to see the status code.

**Comment by ycombinator** [comment]
Hey @terebentina, could you tell us what operation you were trying to perform when you received the `429 Too Many Requests` response?

**Comment by terebentina** [comment]
Hi,

I was doing an index. Something like:
```
_.each(models, function(model) {
	var id = String(model._id);
	esClient.index({
		index: 'estora',
		type: 'contacts',
		id: id,
		body: model.getESData()
	});
});
```
where esClient is the elasticsearch-js client.

**Comment by terebentina** [comment]
May I also suggest a very small code change? I wanted to submit a PR for this but haven't got the time:
Change https://github.com/elastic/elasticsearch-js/blob/master/src/lib/transport.js#L234 to 
```javascript
err = new errors.Generic(body || 'unknown error');
```
This way, if there's any body, you get a more descriptive error. It would also fix #190
